star-rx72n-firmware/lib/rx_spi_comm/src/rx_spi_comm.c

```
@@ -688,49 +839,65 @@ rx_spi_comm_receive(rx_spi_comm_handle_t* handle, rx_frame_t* frame, const uint3
 
-  /* Calculate frame size */
-  const uint32_t header_len = sizeof(header_buf);
-  const uint32_t total_size =
-    k_frame_sync_size + k_frame_header_size + payload_len + k_frame_crc_size;
+    err = internal_read_frame_header(handle, header_buf, &payload_len);
+    if (err != k_rx_ok) {
+      return err;
+    }
+
+    /* Calculate frame size */
+    const uint32_t header_len = sizeof(header_buf);
```

Clang (14.0.6)

[warning] variable 'header_len' is not initialized

(cppcoreguidelines-init-variables)